Documentation/contributing/development/hive.rst

```diff
@@ -0,0 +1,797 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _hive:
+
+Guide to the Hive
+=================
+
+Introduction
+~~~~~~~~~~~~
+
+Cilium is using dependency injection (via ``pkg/hive``) to wire up the
```

Comment:
I would really like to have a one-sentence summary on dependency injection with reflection, and/or a link to learn more on that topic. I don't think all readers will be familiar with the concept.

Reply:
Added both.